includes/connect/class-wc-stripe-connect.php

```diff
@@ -267,6 +267,10 @@ private function save_stripe_keys( $result, $type = 'connect', $mode = 'live' )
 			$options[ $prefix . 'secret_key' ]          = $secret_key;
 			$options[ $prefix . 'connection_type' ]     = $type;
 			$options['pmc_enabled']                     = 'connect' === $type ? 'yes' : 'no'; // When not connected via Connect OAuth, the PMC is disabled.
+			if ( 'connect' === $type && get_option( 'wc_stripe_optimized_checkout_default_on' ) ) {
+				$options['optimized_checkout_element'] = 'yes';
+				delete_option( 'wc_stripe_optimized_checkout_default_on' );
+			}
```

Comment:
I think we should delete the option outside of the connect block, as we won't correctly clean up the option.
```suggestion
			$should_default_optimized_checkout_on = get_option( 'wc_stripe_optimized_checkout_default_on' );
			// Clean up the option.
			delete_option( 'wc_stripe_optimized_checkout_default_on' );
			if ( 'connect' === $type && $should_default_optimized_checkout_on ) {
				$options['optimized_checkout_element'] = 'yes';
			}
```